File: racingmanager-backend/src/main/kotlin/io/github/raginlundf/racingmanager/api/heat/models/HeatModels.kt
package io.github.raginlundf.racingmanager.api.heat.models

import kotlinx.serialization.Serializable

@Serializable
data class HeatResponseModel(
    val id: String,
    val eventId: String,
    val round: Int,
    val heatNumber: Int,
    val status: String,
    val lanes: List<HeatLaneResponseModel>,
    val measurements: List<MeasurementResponseModel>,
    val createdAt: String,
    val armedAt: String? = null,
    val startedAt: String? = null,
    val finishedAt: String? = null,
)

@Serializable
data class HeatLaneResponseModel(
    val lane: Int,
    val participantId: String,
    val participantStartNumber: Int,
    val participantFirstName: String,
    val participantLastName: String,
)

@Serializable
data class MeasurementResponseModel(
    val id: String,
    val heatId: String,
    val lane: Int,
    val durationNanos: Long,
    val outcome: String,
    val receivedAt: String,
)

@Serializable
data class CreateHeatRequestModel(
    val participantIds: List<String>,
)

@Serializable
data class AddMeasurementRequestModel(
    val lane: Int,
    val durationNanos: Long,
    val outcome: String,
)

@Serializable
data class HeatStateChangeEvent(
    val type: String,
    val heat: HeatResponseModel,
)

File: racingmanager-webapp/src/app/races/heat.models.ts
export interface HeatLaneResponse {
  lane: number;
  participantId: string;
  participantStartNumber: number;
  participantFirstName: string;
  participantLastName: string;
}

export interface MeasurementResponse {
  id: string;
  heatId: string;
  lane: number;
  durationNanos: number;
  outcome: string;
  receivedAt: string;
}

export interface HeatResponse {
  id: string;
  eventId: string;
  round: number;
  heatNumber: number;
  status: string;
  lanes: HeatLaneResponse[];
  measurements: MeasurementResponse[];
  createdAt: string;
  armedAt: string | null;
  startedAt: string | null;
  finishedAt: string | null;
}

export interface CreateHeatRequest {
  participantIds: string[];
}

export interface AddMeasurementRequest {
  lane: number;
  durationNanos: number;
  outcome: string;
}

export interface HeatStateChangeEvent {
  type: string;
  heat: HeatResponse;
}
